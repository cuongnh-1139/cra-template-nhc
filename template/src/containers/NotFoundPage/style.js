import styled from 'styled-components';

export const NotFoundPageWrapper = styled.div`
  text-align: center;
  font-weight: bold;
  font-size: 2rem;
  color: ${({ theme }) => theme.palettes.primary};
  padding: ${({ theme }) => theme.sizes.LG};
`;
