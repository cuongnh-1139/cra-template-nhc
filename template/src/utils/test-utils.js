import React from 'react';
import { I18nextProvider } from 'react-i18next';
import { BrowserRouter } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';

import i18n from '../i18n';
import theme from '../theme';
import store from '../stores';

// eslint-disable-next-line
function AppProviders({ children }) {
  return (
    <I18nextProvider i18n={i18n}>
      <BrowserRouter>
        <Provider store={store}>
          <ThemeProvider theme={theme}>
            {children}
          </ThemeProvider>
        </Provider>
      </BrowserRouter>
    </I18nextProvider>
  );
}

const customRender = (ui, options) => render(ui, { wrapper: AppProviders, ...options });

export * from '@testing-library/react';
export { customRender as render };
