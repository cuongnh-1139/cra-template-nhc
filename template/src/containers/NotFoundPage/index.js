import React from 'react';
import { Helmet } from 'react-helmet';
import { useTranslation } from 'react-i18next';
import { NotFoundPageWrapper } from './style';

function NotFoundPage() {
  const { t } = useTranslation();

  return (
    <>
      <Helmet>
        <title>{t('common.notFound404')}</title>
      </Helmet>
      <NotFoundPageWrapper>{t('common.notFound404')}</NotFoundPageWrapper>
    </>
  );
}

export default NotFoundPage;
