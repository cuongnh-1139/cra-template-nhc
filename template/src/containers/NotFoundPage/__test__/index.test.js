import React from 'react';
import { render } from 'utils/test-utils';
import NotFoundPage from '../index';

test('render not found page', () => {
  const { getByText, container } = render(<NotFoundPage />);
  expect(getByText('404 Not Found')).toBeInTheDocument();
  expect(container.firstChild).toMatchSnapshot();
});
