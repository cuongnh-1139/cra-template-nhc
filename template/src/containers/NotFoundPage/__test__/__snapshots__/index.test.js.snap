// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render not found page 1`] = `
.c0 {
  text-align: center;
  font-weight: bold;
  font-size: 2rem;
  color: #676767;
  padding: 32px;
}

<div
  class="c0"
>
  404 Not Found
</div>
`;
